From 117ce77092a19ac877e21a52f50eddd927e10f96 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <adrian.alonso@nxp.com>
Date: Mon, 29 Aug 2016 17:53:23 -0500
Subject: [PATCH 1149/5242] MLK-13184: ARM: dts: imx6q pop arm2 add power key

commit  29546a74420f6ab16f576e7728dd55f1ea052b51 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add power on gpio key for wake up source from suspend

Signed-off-by: Adrian Alonso <adrian.alonso@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6q-pop-arm2.dts |   19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm/boot/dts/imx6q-pop-arm2.dts b/arch/arm/boot/dts/imx6q-pop-arm2.dts
index 764e6a6..df6e2a3 100644
--- a/arch/arm/boot/dts/imx6q-pop-arm2.dts
+++ b/arch/arm/boot/dts/imx6q-pop-arm2.dts
@@ -43,6 +43,19 @@
 		status = "okay";
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_gpio_keys>;
+
+		power {
+			label = "Power Button";
+			gpios = <&gpio3 30 1>;
+			linux,code = <116>;
+			gpio-key,wakeup;
+		};
+	};
+
 	hannstar_cabc {
 		compatible = "hannstar,cabc";
 		lvds_share {
@@ -242,6 +255,12 @@
 			>;
 		};
 
+		pinctrl_gpio_keys: gpio_keysgrp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D30__GPIO3_IO30 		0x1b0b0
+			>;
+		};
+
 		pinctrl_hdmi_cec: hdmicecgrp {
 			fsl,pins = <
 				MX6QDL_PAD_EIM_A25__HDMI_TX_CEC_LINE 0x108b0
-- 
1.7.9.5

